lib/srv/discovery/fetchers/aws-sync/aws-sync.go

```diff
@@ -64,6 +65,8 @@ type Config struct {
 	Integration string
 	// DiscoveryConfigName if set, will be used to report the Discovery Config Status to the Auth Server.
 	DiscoveryConfigName string
+	// Log is the logger to use for logging.
+	Log *slog.Logger
```

Comment:
Since this logger is required, I think you need to add a check to `CheckAndSetDefaults()`